### ISSUE TITLE
🐛 Bug Report: DeprecationError: originalKeywordKind

### ISSUE THREAD

**Issue by drodil** [backstage/backstage#20144]
### 📜 Description

With Typescript 5.2.2 the eslint error is given:
`DeprecationError: 'originalKeywordKind' has been deprecated since v5.0.0 and can no longer be used. Use 'indetifierToKeywordKind(identifier)' instead.`

### 👍 Expected behavior

No error from `typescript/eslint-parser`

### 👎 Actual Behavior with Screenshots

![image](https://github.com/backstage/backstage/assets/1178319/2cb81065-d1da-46cc-a99b-34a7f5fa90e1)

### 👟 Reproduction steps

Update to Typescript 5.2.2.

### 📃 Provide the context for the Bug.

https://stackoverflow.com/questions/76769634/why-am-i-getting-a-deprecationwarning-originalkeywordkind-deprecated-error -> Instead deprecation warning, it's an error due to https://github.com/microsoft/TypeScript/blob/main/src/deprecatedCompat/5.0/identifierProperties.ts#L28

### 🖥️ Your Environment

OS:   Darwin 22.6.0 - darwin/arm64
node: v18.15.0
yarn: 1.22.19
cli:  0.22.13 (installed)
backstage:  1.18.1

Dependencies:
  @backstage/app-defaults                                            1.4.3
  @backstage/backend-app-api                                         0.5.3
  @backstage/backend-common                                          0.17.0, 0.19.5
  @backstage/backend-dev-utils                                       0.1.1
  @backstage/backend-openapi-utils                                   0.0.4
  @backstage/backend-plugin-api                                      0.6.3
  @backstage/backend-tasks                                           0.5.8
  @backstage/backend-test-utils                                      0.2.4
  @backstage/catalog-client                                          1.4.4
  @backstage/catalog-model                                           1.4.2
  @backstage/cli-common                                              0.1.12
  @backstage/cli-node                                                0.1.4
  @backstage/cli                                                     0.22.13
  @backstage/config-loader                                           1.5.0
  @backstage/config                                                  1.1.0
  @backstage/core-app-api                                            1.10.0
  @backstage/core-components                                         0.12.5, 0.13.5
  @backstage/core-plugin-api                                         1.6.0
  @backstage/dev-utils                                               1.0.21
  @backstage/errors                                                  1.2.2
  @backstage/eslint-plugin                                           0.1.3
  @backstage/integration-aws-node                                    0.1.6
  @backstage/integration-react                                       1.1.19
  @backstage/integration                                             1.7.0
  @backstage/plugin-api-docs                                         0.9.11
  @backstage/plugin-app-backend                                      0.3.51
  @backstage/plugin-app-node                                         0.1.3
  @backstage/plugin-auth-backend-module-gcp-iap-provider             0.1.0
  @backstage/plugin-auth-backend-module-github-provider              0.1.0
  @backstage/plugin-auth-backend-module-gitlab-provider              0.1.0
  @backstage/plugin-auth-backend-module-google-provider              0.1.0
  @backstage/plugin-auth-backend-module-oauth2-provider              0.1.0
  @backstage/plugin-auth-backend                                     0.19.0
  @backstage/plugin-auth-node                                        0.2.19, 0.3.0
  @backstage/plugin-catalog-backend-module-aws                       0.2.6
  @backstage/plugin-catalog-backend-module-msgraph                   0.5.10
  @backstage/plugin-catalog-backend-module-scaffolder-entity-model   0.1.0
  @backstage/plugin-catalog-backend-module-unprocessed               0.3.0
  @backstage/plugin-catalog-backend                                  1.13.0
  @backstage/plugin-catalog-common                                   1.0.16
  @backstage/plugin-catalog-graph                                    0.2.36
  @backstage/plugin-catalog-node                                     1.4.4
  @backstage/plugin-catalog-react                                    1.8.4
  @backstage/plugin-catalog-unprocessed-entities                     0.1.3
  @backstage/plugin-catalog                                          1.13.0
  @backstage/plugin-devtools-backend                                 0.2.0
  @backstage/plugin-devtools-common                                  0.1.4
  @backstage/plugin-devtools                                         0.1.4
  @backstage/plugin-entity-feedback-backend                          0.2.0
  @backstage/plugin-entity-feedback-common                           0.1.3
  @backstage/plugin-entity-feedback                                  0.2.7
  @backstage/plugin-events-node                                      0.2.12
  @backstage/plugin-github-actions                                   0.6.5
  @backstage/plugin-home-react                                       0.1.3
  @backstage/plugin-home                                             0.5.8
  @backstage/plugin-jenkins-backend                                  0.2.6
  @backstage/plugin-jenkins-common                                   0.1.19
  @backstage/plugin-jenkins                                          0.8.6
  @backstage/plugin-kubernetes-common                                0.6.6
  @backstage/plugin-org                                              0.6.14
  @backstage/plugin-permission-backend                               0.5.26
  @backstage/plugin-permission-common                                0.7.8
  @backstage/plugin-permission-node                                  0.7.14
  @backstage/plugin-permission-react                                 0.4.15
  @backstage/plugin-proxy-backend                                    0.4.0
  @backstage/plugin-scaffolder-backend-module-confluence-to-markdown 0.2.4
  @backstage/plugin-scaffolder-backend-module-cookiecutter           0.2.27
  @backstage/plugin-scaffolder-backend                               1.17.0
  @backstage/plugin-scaffolder-common                                1.4.1
  @backstage/plugin-scaffolder-node                                  0.2.3
  @backstage/plugin-scaffolder-react                                 1.5.5
  @backstage/plugin-scaffolder                                       1.15.0
  @backstage/plugin-search-backend-module-catalog                    0.1.7
  @backstage/plugin-search-backend-module-elasticsearch              1.3.6
  @backstage/plugin-search-backend-module-techdocs                   0.1.7
  @backstage/plugin-search-backend-node                              1.2.7
  @backstage/plugin-search-backend                                   1.4.3
  @backstage/plugin-search-common                                    1.2.6
  @backstage/plugin-search-react                                     1.7.0
  @backstage/plugin-search                                           1.4.0
  @backstage/plugin-sonarqube-backend                                0.2.5
  @backstage/plugin-sonarqube-react                                  0.1.8
  @backstage/plugin-sonarqube                                        0.7.5
  @backstage/plugin-techdocs-backend                                 1.7.0
  @backstage/plugin-techdocs-module-addons-contrib                   1.1.0
  @backstage/plugin-techdocs-node                                    1.8.0
  @backstage/plugin-techdocs-react                                   1.1.11
  @backstage/plugin-techdocs                                         1.7.0
  @backstage/plugin-user-settings                                    0.7.10
  @backstage/release-manifests                                       0.0.10
  @backstage/test-utils                                              1.4.3
  @backstage/theme                                                   0.2.19, 0.4.2
  @backstage/types                                                   1.1.1
  @backstage/version-bridge                                          1.0.5

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

**Comment by awanlin** [comment]
Hi @drodil, just sharing my experiences on this, they don't feel like a final solution thought. 

I ran into this same error upgrading the Demo site. To fix it I used `5.2.0` for TypeScript and that seemed to work. The other issue I have is that I did not run into this with our instance of Backstage at Keyloop, we upgraded without any issues. We did go to `~5.2.0` about a month ago, I feel like we might have some things in our root `package.json` that are helping with this but I have not had time to dig into it.
